File: app/src/main/java/ru/gilgamesh/abon/motot/payload/response/PageResponse.java
package ru.gilgamesh.abon.motot.payload.response;

import lombok.Getter;
import lombok.Setter;

@Setter@Getter
public class PageResponse {
    //private String pageable;
    public Boolean last;
    private Integer totalPages;
    private Integer totalElements;
    private Integer size;
    private Integer number;
    private Boolean first;
    private Integer numberOfElements;
    private Boolean empty;
}

File: core/src/main/java/ru/gilgamesh/abon/core/data/model/response/PageResponse.kt
package ru.gilgamesh.abon.core.data.model.response

data class PageResponse<T>(
    val content: List<T>,
    val last: Boolean,
    val totalPages: Int,
    val totalElements: Int,
    val size: Int,
    val number: Int,
    val first: Boolean,
    val numberOfElements: Int,
    val empty: Boolean
)